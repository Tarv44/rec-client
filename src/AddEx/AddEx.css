.exchange-form {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.exchange-form > * {
    margin-bottom: 10px;
}

.exchange-form-group {
    flex-direction: column;
    text-align: left;
}

.exchange-form-group input {
    display: block;
    width: 100%;
}

textarea {
    width: 100%;
    height: 100px;
}

#title {
    min-width: 200px;
}