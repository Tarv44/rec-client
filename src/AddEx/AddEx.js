import { Component } from 'react';

export default class AddEx extends Component {
    render() {
        return (
            <main>
                <h2>New Exchange</h2>
            </main>
        )
    }
}