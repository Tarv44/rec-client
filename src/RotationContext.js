import React from 'react';

const RotationContext = React.createContext({
    updateUser: () => {}
})

export default RotationContext;