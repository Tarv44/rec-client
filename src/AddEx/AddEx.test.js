import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import AddEx from './AddEx'
import RotationContext from '../RotationContext'


describe('----- AddEx -----', () => {
    const context = {
        current_user: {
            id: 1,
            username: 'user'
        }
    }
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <RotationContext.Provider value={context}>
                <AddEx />
            </RotationContext.Provider> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <RotationContext.Provider value={context}>
                    <AddEx />
                </RotationContext.Provider>
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})