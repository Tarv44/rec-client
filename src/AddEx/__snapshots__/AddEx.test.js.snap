// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- AddEx ----- renders the UI as expected 1`] = `
<main>
  <form
    autoComplete="off"
    className="exchange-form"
    onSubmit={[Function]}
  >
    <h2>
      New Exchange
    </h2>
    <div
      className="exchange-form-group"
    >
      <label
        htmlFor="add-ex-title"
      >
        Title
      </label>
      <input
        id="add-ex-title"
        name="title"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="exchange-form-group"
    >
      <label
        htmlFor="descr"
      >
        Description
      </label>
      <textarea
        id="descr"
        name="descr"
        onChange={[Function]}
      />
    </div>
    <fieldset
      className="add-ex-song"
    >
      <legend>
        Song
      </legend>
      <div
        className="exchange-form-group"
      >
        <label
          htmlFor="new-song-title-0"
        >
          Title
        </label>
        <input
          id="new-song-title-0"
          name="song-title"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="exchange-form-group"
      >
        <label
          htmlFor="new-song-link-0"
        >
          Link
        </label>
        <input
          id="new-song-link-0"
          name="song-link"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="exchange-form-group"
      >
        <label
          htmlFor="new-artist-0"
        >
          Artist
        </label>
        <input
          id="new-artist-0"
          name="artist"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="exchange-form-group"
      >
        <label
          htmlFor="new-album-0"
        >
          Album
        </label>
        <input
          id="new-album-0"
          name="album"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="exchange-form-group"
      >
        <label
          htmlFor="comment-0"
        >
          Thoughts
        </label>
        <textarea
          className="comment"
          id="comment-0"
          maxLength="1500"
          name="song-title"
          onChange={[Function]}
          value=""
        />
      </div>
    </fieldset>
    <button
      id="add-ex-add-song-btn"
      onClick={[Function]}
    >
      Add Another Song
    </button>
    <button
      id="create-ex-btn"
      type="submit"
    >
      Create Exchange
    </button>
  </form>
</main>
`;
